src/Engine/Utils.java

```diff
@@ -508,6 +508,67 @@ public static void sortLocationsByTravelTime(Unit unit, ArrayList<XYCoord> mapLo
     TravelDistanceComparator tdc = new TravelDistanceComparator(unit, map);
     Collections.sort(mapLocations, tdc);
   }
+  
+
+  /** Returns a string describing the game statistics of a given CO */
+  public static String getGameStatusData(GameMap map, Commander viewed)
+  {
+    StringBuilder sb = new StringBuilder();
+
+    // map state
+    int income = 0;
+    int unitCount = 0;
+    int vehCount = 0;
+    int unitFunds = 0;
+    
+    for( int w = 0; w < map.mapWidth; ++w )
```

Comment:
Instead of looping over every square on the map, why not just iterate through `Commander.ownedProperties` and `Commander.units`?

Reply:
Fog

Reply:
Ah, right. We are potentially looking at other CO's stuff here. Carry on.